### DISCUSSION TITLE
MIT/BSD license?

### DISCUSSION THREAD

**Discussion by davidkhess** [python-sdbus/python-sdbus#51] [General]
Is there any chance this library can be made available under an MIT/BSD style license?

The "relinking" requirement of LGPL is not practical for embedded device applications.

EDIT - meaning, even without modifying this module, making it replaceable with a user provided version of this library isn't really practical.

**Comment by igo95862** [comment]
Hello @davidkhess 

Even if python-sdbus would have been under a permissive license you would still need to link against `libsystemd` which is licensed under LGPL v2.1 or later.

Also aren't the Python modules technically a dynamic linking already? I don't know how Python modules work for embedded devices.

**Comment by davidkhess** [comment]
That's true enough re: libsystemd. Hand-waving that away for the moment....

I'm not worried about viral / poisoning issues with the proprietary code. I'm concerned with the requirement to make it possible for users to replace any components with an LGPL license. I suppose in this scenario you'd have to make it possible for people to upload and overwrite the stock copy of this module with one provided by them.

I.e. it's the logistics that start to get tricky whenever everything has to be packaged for consumer grade devices and the LGPL requires allowing end users to replace LGPL'd components.

**Comment by souliane** [comment]
Hi, can you please clarify why there are both LGPL and GPL licenses files in the project? The whole code looks LGPL though.
Thanks!

**Reply by igo95862** [reply]
Hello @souliane.

This is because the LGPL license is an extension of GPL so both licenses' texts are required. See #40 .

**Reply by igo95862** [reply]
I should probably add a section to README explaining this because too many people have this question.
